### PR TITLE
[TASK] Add signal documentation

### PR DESCRIPTION
**Pull request by psychomieze**

### CODE REVIEW

File: Documentation/ApiOverview/Hooks/Concept/Index.rst [hunk 1]
```diff
@@ -7,35 +7,38 @@
...
+be executed. However, hooks and signals have to be implemented in the TYPO3 core
 before you can use them, while extending a PHP class via the XCLASS
 method allows you to extend any class you like.
 
 .. _hooks-proposing:
 
-Proposing hooks
-"""""""""""""""
+Proposing hooks or signals
+"""""""""""""""""""""""""""
```
Comment (liayn) [inline Documentation/ApiOverview/Hooks/Concept/Index.rst]: one quote too much?

File: Documentation/ApiOverview/Hooks/Concept/Index.rst [hunk 2]
```diff
@@ -97,3 +100,43 @@ API documentation of :code:`\TYPO3\CMS\Core\Utility\GeneralUtility`.
...
+
+.. code-block:: php
+
+   $this->signalSlotDispatcher->dispatch(__CLASS__, 'afterExtensionUninstall', [$extensionKey, $this]);
+
+In this case, the :code:`dumpClassLoadingInformation` method will get the extension key and an instance of
+the dispatching class as parameters.
+
+Finding Signals
+""""""""""""""""
```
Comment (liayn) [inline Documentation/ApiOverview/Hooks/Concept/Index.rst]: same here

File: Documentation/ApiOverview/Hooks/Concept/Index.rst [hunk 3]
```diff
@@ -97,3 +100,43 @@ API documentation of :code:`\TYPO3\CMS\Core\Utility\GeneralUtility`.
...
+
+There is no complete list of signals available, but they are easily found by searching the TYPO3
```
Comment (liayn) [inline Documentation/ApiOverview/Hooks/Concept/Index.rst]: ..., ut they can be easily found

**Comment by psychomieze** [comment]
Thx @liayn, fixed :)

**Comment by marble** [comment]
Great!
Some reST tips for my beloved writers:
The first headline is the document title. It makes sense to make it special. Usually: '===' above AND under.
The scheme of headline markup is independent of the other files. A common hierarchy is '===', '---', '~~~'
There was one error: With a trailing underscore you can ONLY link to targets within the SAME file. That's a Docutils reST feature. To link to a target within the same PROJECT you have to use Sphinx "cross-referencing": `:ref:'hooks-configuration'` (should be back-ticks, can't show that here).

Technically linelength doesn't matter in reST. For a TYPO3 reST style guide I'll recommend a maximum of 79 characters in a line. This is not strict. But allows side by side comparisons and editing in web forms.

**Comment by marble** [comment]
Ah, one more:
- Longer underlining of headlines works.
- _Slightly_ longer underlinings look ugly. The look like an error. But they work.
- Too short underlinings do not work
- For over- and underlining they must be of the same length.

**Comment by liayn** [comment]
@marble thanks for the heads up. duly noted.

**Comment by psychomieze** [comment]
@marble thanks for the reformatting and the info :) I'll hopefully someday learn how to write rst.

**Comment by marble** [comment]
:-))
 Alright, so I know you're keeping track of it and I don't have to.
And, about reST: I'm sure you're doing well. That's why I think you may appreciate a tip. What can go wrong? It's "docs". You can't break everybody's code and have everybody hunting you as with code. For me it's great to see "one more" starts "doc hacking" :-)) Don't hesitate to just try and DO WRITE samething. I'm pleased to "reSTify it".

**Pull request merged**